test/unclogged/core_test.clj

```diff
@@ -209,3 +210,44 @@
       :rfc-5424 MessageFormat/RFC_5424
       :rfc_5424 MessageFormat/RFC_5424
       :rfc5424 MessageFormat/RFC_5424)))
+
+(deftest ->syslog-msg-tests
+  (testing "no defaults, all keys"
+    (let [contents {:message "hello"
+                    :message-id "xyzzy"
+                    :app-name "unclogged"
+                    :hostname "ditka"
+                    :process-id "1234"}
+          syslog-msg (@#'unclogged.core/->syslog-msg {} contents)]
+      (is (= "hello" (.toString ^CharArrayWriter
+                                (.getMsg ^SyslogMessage syslog-msg))))
+      (is (= "xyzzy" (.getMsgId ^SyslogMessage syslog-msg)))
+      (is (= "unclogged" (.getAppName ^SyslogMessage syslog-msg)))
+      (is (= "ditka" (.getHostname ^SyslogMessage syslog-msg)))
+      (is (= "1234" (.getProcId ^SyslogMessage syslog-msg)))))
+  (testing "some defaults, all keys"
+    (let [defaults {:app-name "unclogged"
+                    :hostname "ditka"
+                    :process-id "1234"}
+          contents {:message "hello"
+                    :message-id "xyzzy"}
+          syslog-msg (@#'unclogged.core/->syslog-msg defaults contents)]
```

Comment:
You're doing the right thing, apparently. Unless you explicitly want the early deref, calling a var works because of a happy accident in how some internals work, and is actually preferred. Rich ipse dixit: https://groups.google.com/forum/#!msg/clojure/Mi277rszUs0/G9Q0ua9nM1EJ